src/dbnode/storage/block/block.go

```diff
@@ -230,7 +230,11 @@ func (b *dbBlock) Stream(blocker context.Context) (xio.BlockReader, error) {
 	lockUpgraded = true
 	b.Lock()
 
-	// NB: need to re-check as we just upgraded the lock.
+	// Need to re-check everything since we upgraded the lock.
+	if b.closed {
```

Comment:
can you add a test case for this too

Reply:
Should I just do a concurrency test? I can't think of any other way...

Reply:
yeah it'll have to be. can do the same thing we do in the shard where we have a `testXYZ` that kicks off two go routines calling the different methods, and run `testXYZ` a bunch of times